src/runtime/Symbol.cpp

```diff
@@ -47,13 +58,127 @@ Symbol* Symbol::fromGlobalSymbolRegistry(VMInstance* vm, String* stringKey)
     return newSymbol;
 }
 
+Value Symbol::keyForSymbol(VMInstance* vm, Symbol* sym)
+{
+    ASSERT(!!sym);
+    auto& list = vm->globalSymbolRegistry();
+
+    for (size_t i = 0; i < list.size(); i++) {
+        // If SameValue(e.[[Symbol]], sym) is true, return e.[[Key]].
+        if (list[i].symbol == sym) {
```

Comment:
Maybe it is defined in the specification, but is it true that always maximum of one result is possible?

Reply:
Yes, only one unique Symbol (with each Symbol description) will be registered according to the spec.